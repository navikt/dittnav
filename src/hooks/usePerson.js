import { useQuery } from 'react-query';
import { fetchOppfolging, fetchMeldekort, fetchMeldinger, fetchPersonIdent, fetchPersonNavn } from '../Api';
import useStore from './useStore';

export const useOppfolging = () => (
  [useQuery('oppfolging', fetchOppfolging, { onError: useStore().setError })]
);

export const useMeldekort = () => (
  [useQuery('meldekort', fetchMeldekort, { onError: useStore().setError })]
);

export const useMeldinger = () => (
  [useQuery('meldinger', fetchMeldinger, { onError: useStore().setError })]
);

export const useIdent = (enabled) => (
  [useQuery('ident', fetchPersonIdent, { enabled, onError: useStore().setError })]
);

export const useNavn = () => (
  [useQuery('navn', fetchPersonNavn, { onError: useStore().setError })]
);

const usePerson = () => (
  [useOppfolging(), useMeldekort(), useMeldinger(), useNavn()]
);

export default usePerson;
