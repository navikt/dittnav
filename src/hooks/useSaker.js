import { useQuery } from 'react-query';
import { fetchSakstema, fetchSaker } from '../Api';
import useStore from './useStore';

export const useSakstema = () => (
  [useQuery('sakstema', fetchSakstema, { onError: useStore().setError })]
);

export const usePaabegynteSoknader = () => {
  const store = useStore();

  const checkForFeilendeBaksystem = (data) => {
    if (data.content.feilendeBaksystem.length > 0) {
      store.setError();
    }
  };

  return [useQuery('paabegynteSoknader', fetchSaker, { onSuccess: checkForFeilendeBaksystem, onError: store.setError })];
};

const useSaker = () => (
  [useSakstema(), usePaabegynteSoknader()]
);

export default useSaker;
