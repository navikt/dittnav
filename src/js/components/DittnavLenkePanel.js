import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from 'react-intl';
import { Undertekst } from 'nav-frontend-typografi';
import Config from '../Config';
import { LenkepanelMedIkon, IkonKane, IkonPille, IkonSkilt } from './LenkepanelMedIkon';
import OversiktspanelMedListe from './OversiktspanelMedListe';
import DinesakerSakstema from './DinesakerSakstema';

// TODO:    - finn svg data for ikoner

const MAX_SAKER_SOM_VISES = 2;

const saksoversiktUrl = `${Config.dittNav.SERVICES_URL}/saksoversikt/`;
const utbetalingsoversiktUrl = `${Config.dittNav.SERVICES_URL}/utbetalingsoversikt/`;
const innboksUrl = `${Config.dittNav.SERVICES_URL}/mininnboks/`;

class DittnavLenkePanel extends React.Component {

  makeSaksoversiktPanel(sakstemaList) {
    const alt = "Dine saker";
    const overskrift = "fliser.dine.saker";

    if (sakstemaList) {
      const { antallSakstema } = this.props.sakstema;
      const numRemainingSaker = antallSakstema - sakstemaList.length;
      const showFooter = numRemainingSaker <= 0 && sakstemaList.length < MAX_SAKER_SOM_VISES;

      const footer = showFooter ?
        <div key="footer">
          <hr />
          <Undertekst>
            <FormattedMessage id="saksoversikt.ingen.flere.saker" />
          </Undertekst>
        </div> : null;

      const liste = sakstemaList.map((tema) => (
        <DinesakerSakstema
          key={tema.temanavn}
          dato={tema.sisteOppdatering}
          status={tema.sisteBehandlingStatus}
          temanavn={tema.temanavn}
          href={tema.url}
        />
      )).concat(footer);

      return (
        <OversiktspanelMedListe
          className="dittnav-lenkepanel-stor"
          alt={alt}
          overskrift={<FormattedMessage id={overskrift} />}
          ikon={<IkonSkilt />}
          headerLenkeTekst={<FormattedMessage id="saksoversikt.alle.saker" values={{ count: antallSakstema }} />}
          headerLenkeHref={saksoversiktUrl}
          border
          liste={liste}
        />
        );
    }
    else {
      return (
        <LenkepanelMedIkon
          className="dittnav-lenkepanel-smaa-item"
          alt={alt}
          overskrift={overskrift}
          ingress=""
          href={saksoversiktUrl}
        >
          <IkonSkilt />
        </LenkepanelMedIkon>
      );
    }
  }

  render() {
    const { sakstemaList } = this.props.sakstema;
    const sakstemaListPruned = sakstemaList && sakstemaList.length > 0
      ? sakstemaList.slice(0, MAX_SAKER_SOM_VISES) : null;
    const saksoversiktPanel = this.makeSaksoversiktPanel(sakstemaListPruned);

    return (
      <div className="dittnav-lenkepanel-top-container">
        { sakstemaListPruned ? saksoversiktPanel : null }
        <div className="dittnav-lenkepanel-smaa" id={ sakstemaListPruned ? "cols-layout" : null }>
          { !sakstemaListPruned ? saksoversiktPanel : null }
          <LenkepanelMedIkon
            alt="Utbetalinger"
            overskrift="fliser.dine.utbetalinger"
            ingress=""
            className="dittnav-lenkepanel-smaa-item"
            href={utbetalingsoversiktUrl}
          >
            <IkonPille />
          </LenkepanelMedIkon>
          <LenkepanelMedIkon
            alt="Innboks"
            overskrift="fliser.innboks"
            ingress=""
            className="dittnav-lenkepanel-smaa-item last"
            href={innboksUrl}
          >
            <IkonKane />
          </LenkepanelMedIkon>
        </div>
      </div>
    );
  }
}

DittnavLenkePanel.propTypes = {
  sakstema: PropTypes.shape({
    antallSakstema: PropTypes.number.isRequired,
    sakstemaList: PropTypes.arrayOf(PropTypes.shape({
      temanavn: PropTypes.string.isRequired,
      sisteBehandlingStatus: PropTypes.string.isRequired,
      sisteOppdatering: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
    })).isRequired,
  }).isRequired,
};

export default DittnavLenkePanel;
