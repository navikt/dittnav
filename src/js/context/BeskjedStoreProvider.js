import React, { createContext, useReducer } from 'react';
import { arrayOf, node } from 'prop-types';
import Api from '../Api';
import {
  ADD_BESKJEDER,
  ADD_INAKTIVE_BESKJEDER,
  ADD_INAKTIV_BESKJED, REMOVE_BESKJED,
} from '../types/Actions';
import BeskjedType from '../types/BeskjedType';

const initialState = (beskjeder, inaktiveBeskjeder) => ({
  beskjeder,
  inaktiveBeskjeder,
});

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_BESKJEDER:
      return {
        ...state,
        beskjeder: action.payload,
      };
    case ADD_INAKTIVE_BESKJEDER:
      return {
        ...state,
        inaktiveBeskjeder: action.payload,
      };
    case REMOVE_BESKJED:
      Api.postDone({
        eventId: action.payload.eventId,
        uid: action.payload.uid,
      });
      return {
        ...state,
        beskjeder: state.beskjeder.filter(b => action.payload.uid !== b.uid),
      };
    case ADD_INAKTIV_BESKJED:
      return {
        ...state,
        inaktiveBeskjeder: [...state.inaktiveBeskjeder, action.payload],
      };
    default:
      return state;
  }
};

export const BeskjedStoreContext = createContext(null);

const BeskjedStoreProvider = ({ beskjeder, inaktiveBeskjeder, children }) => {
  const [state, dispatch] = useReducer(reducer, initialState(beskjeder, inaktiveBeskjeder));
  const value = { state, dispatch };

  return (
    <BeskjedStoreContext.Provider value={value}>
      {children}
    </BeskjedStoreContext.Provider>
  );
};

BeskjedStoreProvider.propTypes = {
  beskjeder: arrayOf(BeskjedType),
  inaktiveBeskjeder: arrayOf(BeskjedType),
  children: node.isRequired,
};

BeskjedStoreProvider.defaultProps = {
  beskjeder: null,
  inaktiveBeskjeder: null,
};

export default BeskjedStoreProvider;
