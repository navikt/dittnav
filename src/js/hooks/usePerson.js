import { useQuery } from 'react-query';
import Api from '../Api';

export const useOppfolging = () => (
  [useQuery('oppfolging', Api.fetchOppfolging)]
);

export const useMeldekort = () => (
  [useQuery('meldekort', Api.fetchMeldekort)]
);

export const useMeldinger = () => (
  [useQuery('meldinger', Api.fetchMeldinger)]
);

export const useNavn = () => (
  [useQuery('navn', Api.fetchPersonNavn)]
);

export const useIdent = () => (
  [useQuery('ident', Api.fetchPersonIdent)]
);
