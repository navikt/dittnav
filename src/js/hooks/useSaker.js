import { useQuery } from 'react-query';
import Api from '../Api';
import useStore from './useStore';

export const useSakstema = () => (
  [useQuery('sakstema', Api.fetchSakstema, { onError: useStore().setError })]
);

export const usePaabegynteSoknader = () => (
  [useQuery('paabegynteSoknader', Api.fetchSaker, { onError: useStore().setError })]
);

const useSaker = () => (
  [useSakstema(), usePaabegynteSoknader()]
);

export default useSaker;
