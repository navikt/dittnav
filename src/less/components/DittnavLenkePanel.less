@import (reference) '~nav-frontend-core/less/_variabler';

.dittnav-lenkepanel-top-container {
  display: flex;
  flex-direction: row;
  margin-bottom: 1em;

  .dittnav-lenkepanel-stor {
    flex: 2 0 auto;
    box-sizing: border-box;
    margin-right: 1em;
    margin-bottom: 1em;
  }

  .dittnav-lenkepanel-liten {
    display: flex;
    flex-direction: row;
    align-content: space-between;
    flex: 1 1 auto;

    .dittnav-lenkepanel-liten-item {
      flex: 1 1 auto;
      box-sizing: border-box;
      margin-right: 1em;
      margin-bottom: 1em;
      width: 100%;

      .lenkepanel__heading {
        padding: 0.5em;
      }
    }

    .last {
      margin-right: 0;
    }

    &#cols-layout {
      flex-direction: column;
      margin-right: 0;
    }
  }

  @media (max-width: @screen-sm-min) {
    flex-direction: column;
    .dittnav-lenkepanel-stor {
      margin-right: 0;
    }
    .dittnav-lenkepanel-liten {
      flex-direction: column;
    }
  }

  .lenke-overstyring() {
    border-bottom: 0;
    &:not(:hover) {
      text-decoration: none;
    }
  }

  #oversiktspanel__header-lenke-id, #sak-lenke-id {
    .lenke-overstyring();
  }
}
