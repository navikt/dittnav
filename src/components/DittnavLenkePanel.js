import React from 'react';
import { FormattedMessage as F } from 'react-intl';
import Lenkepanel from 'nav-frontend-lenkepanel/lib';
import OversiktspanelMedListe from './common/OversiktspanelMedListe';
import DinesakerSakstema from './DinesakerSakstema';
import { listOfActions, listOfComponentNames, logAmplitudeEvent } from '../utils/amplitudeUtils';
import { lenker } from '../utils/lenker';
import { useSakstema } from '../hooks/useSaker';

const antallSakstemaVist = 2;

const DittnavLenkePanel = () => {
  const [{ data: saker }] = useSakstema();
  const visStortSakspanel = saker && saker.content.sakstemaer?.length > 0;
  const sakerURL = saker?.content?.sakerURL;

  console.log(sakerURL)

  return (
    <div className="dittnav-lenkepanel-top-container">
      {visStortSakspanel
        ? (
          <OversiktspanelMedListe
            className="dittnav-lenkepanel-stor"
            overskrift={<F id="saksoversikt.overskrift" />}
            headerLenkeTekst={<F id="saksoversikt.alle.saker" />}
            headerLenkeHref={sakerURL}
            border={false}
            liste={
              saker.content.sakstemaer
                .slice(0, antallSakstemaVist)
                .map((sak) => (
                  <DinesakerSakstema
                    key={sak.kode}
                    tema={sak}
                  />
                ))
            }
          />
        ) : null}
      <div className="dittnav-lenkepanel-liten" id={visStortSakspanel ? 'cols-layout' : null}>
        {!visStortSakspanel
          ? (
            <Lenkepanel
              alt="Dine saker"
              className="dittnav-lenkepanel-liten-item"
              onClick={() => logAmplitudeEvent(listOfComponentNames.DineSisteSakerFlis, listOfActions.TrykkPaaLenke)}
              href={saker?.content.sakerURL}
              border
            >
              <F id="fliser.dine.saker" />
            </Lenkepanel>
          ) : null}
        <Lenkepanel
          alt="Utbetalinger"
          className="dittnav-lenkepanel-liten-item"
          href={lenker.utbetalingsoversikt.url}
          onClick={() => logAmplitudeEvent(listOfComponentNames.UtbetalingerFlis, listOfActions.TrykkPaaLenke)}
          border
        >
          <F id="fliser.dine.utbetalinger" />
        </Lenkepanel>
        <Lenkepanel
          alt="Innboks"
          className="dittnav-lenkepanel-liten-item last"
          href={lenker.innboks.url}
          onClick={() => logAmplitudeEvent(listOfComponentNames.InnboksFlis, listOfActions.TrykkPaaLenke)}
          border
        >
          <F id="fliser.innboks" />
        </Lenkepanel>
      </div>
    </div>
  );
};

export default DittnavLenkePanel;
