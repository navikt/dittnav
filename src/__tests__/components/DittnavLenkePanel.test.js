import * as React from 'react';
import sinon from 'sinon';
import wrapIntl from 'utils/intlTestHelper';
import DittnavLenkePanel from 'components/DittnavLenkePanel';
import BeskjedStoreProvider from 'context/StoreProvider';
import * as useSaker from '../../hooks/useSaker';
import * as usePerson from '../../hooks/usePerson';

const ReactTestRenderer = require('react-test-renderer');

jest.mock('react-ga');
let sandbox = null;

beforeEach(() => {
  sandbox = sinon.createSandbox();
});

afterEach(() => {
  useSaker.useSakstema.restore();
  usePerson.useOppfolging.restore();
});

const sakstemaMedSaker = {
  antallSakstema: 6,
  sakstemaList: [
    {
      temanavn: 'Barnetrygd',
      temakode: 'BAR',
      antallStatusUnderBehandling: 0,
      sisteOppdatering: '2019-08-15T12:00:00+02:00',
    },
    {
      temanavn: 'Sykepenger og oppfÃ¸lging',
      temakode: 'SYK',
      antallStatusUnderBehandling: 10,
      sisteOppdatering: '2019-08-14T12:00:00+02:00',
    },
  ],
};

const sakstemaUtenSaker = {
  antallSakstema: 0,
  sakstemaList: [],
};

test('Snapshot test med saker', () => {
  sandbox.stub(useSaker, 'useSakstema').returns([
    {
      data: { content: sakstemaMedSaker },
      isLoading: false,
      isSuccess: true,
    },
  ]);

  sandbox.stub(usePerson, 'useOppfolging').returns([
    {
      data: { content: { erBrukerUnderOppfolging: false } },
      isLoading: false,
      isSuccess: true,
    },
  ]);

  const component = ReactTestRenderer.create(wrapIntl(
    <BeskjedStoreProvider>
      <DittnavLenkePanel />
    </BeskjedStoreProvider>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('Snapshot test uten saker', () => {
  sandbox.stub(useSaker, 'useSakstema').returns([
    {
      data: { content: sakstemaUtenSaker },
      isLoading: false,
      isSuccess: true,
    },
  ]);

  sandbox.stub(usePerson, 'useOppfolging').returns([
    {
      data: { content: { erBrukerUnderOppfolging: false } },
      isLoading: false,
      isSuccess: true,
    },
  ]);

  const component = ReactTestRenderer.create(wrapIntl(
    <BeskjedStoreProvider>
      <DittnavLenkePanel />
    </BeskjedStoreProvider>,
  ));

  expect(component.toJSON()).toMatchSnapshot();
});
