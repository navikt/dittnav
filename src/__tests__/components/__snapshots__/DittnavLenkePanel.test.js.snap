// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test med saker 1`] = `
<div
  className="dittnav-lenkepanel-top-container"
>
  <div
    className="panel oversiktspanel dittnav-lenkepanel-stor panel--border"
  >
    <div
      className="oversiktspanel__header"
    >
      <div
        className="oversiktspanel__overskrift"
      >
        <div
          className="lenkepanel__ikon"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 177 177"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M93 164H85V17C85.5 16 87 14 89 14C91 14 92.6667 16 93 17V164Z"
              fill="#6D655F"
            />
            <path
              d="M144 130V85H31L15 107.5L31 130H144Z"
              fill="#90C9A7"
            />
            <path
              d="M34 74V29H147L163 51.5L147 74H34Z"
              fill="#FFB45B"
            />
          </svg>
        </div>
        <h2
          className="typo-undertittel"
        >
          <span>
            Dine siste saker
          </span>
        </h2>
      </div>
      <p
        className="typo-normal oversiktspanel__header-lenketekst"
      >
        <a
          className="oversiktspanel__header-lenke"
          href="http://localhost:9222/saksoversikt/"
          id="dekorator-bottomborder-overstyring"
        >
          <span>
            Se alle dine saker (5)
          </span>
          <i
            className="nav-frontend-chevron chevronboks chevron--hoyre"
            id="oversiktspanel__chevron"
          />
        </a>
      </p>
    </div>
    <div>
      <hr />
    </div>
    <div
      className="oversiktspanel__liste"
    >
      <div
        className="sak-container"
      >
        <a
          className="sak-lenke"
          href="#"
          id="dekorator-bottomborder-overstyring"
        >
          <div
            className="sak-temanavn"
          >
            <span
              className="typo-undertittel"
            >
              Tema 1
            </span>
            <i
              className="nav-frontend-chevron chevronboks sak-chevron chevron--hoyre"
            />
          </div>
          <div
            className="sak-dato"
          >
            <span
              className="typo-undertekst"
            >
              <span>
                Sist oppdatert
              </span>
               
              <span>
                2019 M07 22
              </span>
               - 
              <span>
                Ferdig behandlet
              </span>
            </span>
          </div>
        </a>
      </div>
      <div
        className="sak-container"
      >
        <a
          className="sak-lenke"
          href="#"
          id="dekorator-bottomborder-overstyring"
        >
          <div
            className="sak-temanavn"
          >
            <span
              className="typo-undertittel"
            >
              Tema 2
            </span>
            <i
              className="nav-frontend-chevron chevronboks sak-chevron chevron--hoyre"
            />
          </div>
          <div
            className="sak-dato"
          >
            <span
              className="typo-undertekst"
            >
              <span>
                Sist oppdatert
              </span>
               
              <span>
                2019 M07 23
              </span>
               - 
              <span>
                Under behandling
              </span>
            </span>
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="dittnav-lenkepanel-smaa"
    id="cols-layout"
  >
    <a
      className="lenkepanel dittnav-lenkepanel-smaa-item lenkepanel--border"
      href="http://localhost:9222/utbetalingsoversikt/"
      onClick={null}
    >
      <div
        className="lenkepanel__innhold lenkepanel__heading"
      >
        <div
          className="lenkepanel__ikon"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 177 177"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M62.6616 63.4795L114.327 115.591L145.089 84.4894C159.338 70.0829 159.21 46.853 144.804 32.6039C130.397 18.3548 107.167 18.4824 92.9181 32.8889L62.6616 63.4795Z"
              fill="#65A9C1"
              fillRule="evenodd"
            />
            <path
              d="M87.5801 58.3952C86.8022 59.1794 86.8074 60.4457 87.5916 61.2236C88.3758 62.0015 89.6421 61.9963 90.42 61.2121L87.5801 58.3952ZM103.989 44.6926L105.409 46.101C105.413 46.0969 105.417 46.0927 105.421 46.0885L103.989 44.6926ZM118.011 38.8431L117.856 40.8371L118.011 38.8431ZM131.665 47.1569C132.488 47.894 133.752 47.8248 134.49 47.0021C135.227 46.1795 135.157 44.9151 134.335 44.178L131.665 47.1569ZM90.42 61.2121L105.409 46.101L102.569 43.2841L87.5801 58.3952L90.42 61.2121ZM103.989 44.6926C105.421 46.0885 105.421 46.0889 105.421 46.0893C105.42 46.0894 105.42 46.0898 105.42 46.09C105.419 46.0905 105.419 46.0908 105.419 46.091C105.418 46.0915 105.418 46.0916 105.419 46.0913C105.419 46.0907 105.421 46.0886 105.425 46.0851C105.432 46.078 105.445 46.0651 105.463 46.0469C105.5 46.0104 105.56 45.9524 105.642 45.8761C105.805 45.7235 106.053 45.4986 106.377 45.227C107.027 44.6819 107.971 43.9582 109.135 43.2535C111.494 41.8251 114.573 40.5824 117.856 40.8371L118.166 36.849C113.713 36.5037 109.781 38.1858 107.063 39.8321C105.688 40.6648 104.577 41.516 103.807 42.1614C103.421 42.485 103.118 42.7593 102.907 42.9571C102.801 43.056 102.719 43.136 102.66 43.1937C102.63 43.2225 102.607 43.2458 102.59 43.2631C102.581 43.2717 102.574 43.2789 102.569 43.2845C102.566 43.2873 102.564 43.2897 102.562 43.2917C102.561 43.2928 102.56 43.2937 102.559 43.2945C102.558 43.2949 102.558 43.2954 102.558 43.2956C102.557 43.2962 102.557 43.2966 103.989 44.6926ZM117.856 40.8371C123.592 41.2819 127.546 43.4655 131.665 47.1569L134.335 44.178C129.751 40.0701 125.001 37.3791 118.166 36.849L117.856 40.8371Z"
              fill="#DCF3FA"
            />
            <path
              clipRule="evenodd"
              d="M115.031 114.915L63.3652 62.8037L32.6039 93.9048C18.3548 108.311 18.4824 131.541 32.8889 145.79C47.2954 160.039 70.5253 159.912 84.7744 145.505L115.031 114.915Z"
              fill="#D5D98E"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div>
          <h2
            className="typo-undertittel"
          >
            <span>
              Dine utbetalinger
            </span>
          </h2>
          
        </div>
      </div>
      <span
        className="lenkepanel__indikator"
      />
    </a>
    <a
      className="lenkepanel dittnav-lenkepanel-smaa-item last lenkepanel--border"
      href="http://localhost:9222/mininnboks/"
      onClick={null}
    >
      <div
        className="lenkepanel__innhold lenkepanel__heading"
      >
        <div
          className="lenkepanel__ikon"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 177 177"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M116 164V172"
              stroke="#4F4742"
              strokeLinecap="round"
              strokeWidth="10"
            />
            <path
              d="M116 32H121V31.6316L120.946 31.2673L116 32ZM121 167V32H111V167H121ZM120.946 31.2673C120.23 26.4341 118.44 18.7426 113.587 12.1585C108.545 5.31666 100.414 0 88 0V10C97.0862 10 102.289 13.6833 105.538 18.0915C108.977 22.7574 110.437 28.5659 111.054 32.7327L120.946 31.2673ZM88 0C75.0773 0 67.2216 6.82543 62.7924 14.991C58.5379 22.8345 57.3567 31.9633 57.0092 37.6975L66.9908 38.3025C67.31 33.0367 68.3787 25.6655 71.5826 19.759C74.6117 14.1746 79.4227 10 88 10V0Z"
              fill="#A96215"
            />
          </svg>
        </div>
        <div>
          <h2
            className="typo-undertittel"
          >
            <span>
              Innboks
            </span>
          </h2>
          
        </div>
      </div>
      <span
        className="lenkepanel__indikator"
      />
    </a>
  </div>
</div>
`;

exports[`Snapshot test uten saker 1`] = `
<div
  className="dittnav-lenkepanel-top-container"
>
  <div
    className="dittnav-lenkepanel-smaa"
    id={null}
  >
    <a
      className="lenkepanel dittnav-lenkepanel-smaa-item lenkepanel--border"
      href="http://localhost:9222/saksoversikt/"
      onClick={null}
    >
      <div
        className="lenkepanel__innhold lenkepanel__heading"
      >
        <div
          className="lenkepanel__ikon"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 177 177"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M93 164H85V17C85.5 16 87 14 89 14C91 14 92.6667 16 93 17V164Z"
              fill="#6D655F"
            />
            <path
              d="M144 130V85H31L15 107.5L31 130H144Z"
              fill="#90C9A7"
            />
            <path
              d="M34 74V29H147L163 51.5L147 74H34Z"
              fill="#FFB45B"
            />
          </svg>
        </div>
        <div>
          <h2
            className="typo-undertittel"
          >
            <span>
              Dine saker
            </span>
          </h2>
          
        </div>
      </div>
      <span
        className="lenkepanel__indikator"
      />
    </a>
    <a
      className="lenkepanel dittnav-lenkepanel-smaa-item lenkepanel--border"
      href="http://localhost:9222/utbetalingsoversikt/"
      onClick={null}
    >
      <div
        className="lenkepanel__innhold lenkepanel__heading"
      >
        <div
          className="lenkepanel__ikon"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 177 177"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M62.6616 63.4795L114.327 115.591L145.089 84.4894C159.338 70.0829 159.21 46.853 144.804 32.6039C130.397 18.3548 107.167 18.4824 92.9181 32.8889L62.6616 63.4795Z"
              fill="#65A9C1"
              fillRule="evenodd"
            />
            <path
              d="M87.5801 58.3952C86.8022 59.1794 86.8074 60.4457 87.5916 61.2236C88.3758 62.0015 89.6421 61.9963 90.42 61.2121L87.5801 58.3952ZM103.989 44.6926L105.409 46.101C105.413 46.0969 105.417 46.0927 105.421 46.0885L103.989 44.6926ZM118.011 38.8431L117.856 40.8371L118.011 38.8431ZM131.665 47.1569C132.488 47.894 133.752 47.8248 134.49 47.0021C135.227 46.1795 135.157 44.9151 134.335 44.178L131.665 47.1569ZM90.42 61.2121L105.409 46.101L102.569 43.2841L87.5801 58.3952L90.42 61.2121ZM103.989 44.6926C105.421 46.0885 105.421 46.0889 105.421 46.0893C105.42 46.0894 105.42 46.0898 105.42 46.09C105.419 46.0905 105.419 46.0908 105.419 46.091C105.418 46.0915 105.418 46.0916 105.419 46.0913C105.419 46.0907 105.421 46.0886 105.425 46.0851C105.432 46.078 105.445 46.0651 105.463 46.0469C105.5 46.0104 105.56 45.9524 105.642 45.8761C105.805 45.7235 106.053 45.4986 106.377 45.227C107.027 44.6819 107.971 43.9582 109.135 43.2535C111.494 41.8251 114.573 40.5824 117.856 40.8371L118.166 36.849C113.713 36.5037 109.781 38.1858 107.063 39.8321C105.688 40.6648 104.577 41.516 103.807 42.1614C103.421 42.485 103.118 42.7593 102.907 42.9571C102.801 43.056 102.719 43.136 102.66 43.1937C102.63 43.2225 102.607 43.2458 102.59 43.2631C102.581 43.2717 102.574 43.2789 102.569 43.2845C102.566 43.2873 102.564 43.2897 102.562 43.2917C102.561 43.2928 102.56 43.2937 102.559 43.2945C102.558 43.2949 102.558 43.2954 102.558 43.2956C102.557 43.2962 102.557 43.2966 103.989 44.6926ZM117.856 40.8371C123.592 41.2819 127.546 43.4655 131.665 47.1569L134.335 44.178C129.751 40.0701 125.001 37.3791 118.166 36.849L117.856 40.8371Z"
              fill="#DCF3FA"
            />
            <path
              clipRule="evenodd"
              d="M115.031 114.915L63.3652 62.8037L32.6039 93.9048C18.3548 108.311 18.4824 131.541 32.8889 145.79C47.2954 160.039 70.5253 159.912 84.7744 145.505L115.031 114.915Z"
              fill="#D5D98E"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div>
          <h2
            className="typo-undertittel"
          >
            <span>
              Dine utbetalinger
            </span>
          </h2>
          
        </div>
      </div>
      <span
        className="lenkepanel__indikator"
      />
    </a>
    <a
      className="lenkepanel dittnav-lenkepanel-smaa-item last lenkepanel--border"
      href="http://localhost:9222/mininnboks/"
      onClick={null}
    >
      <div
        className="lenkepanel__innhold lenkepanel__heading"
      >
        <div
          className="lenkepanel__ikon"
        >
          <svg
            fill="none"
            height="48"
            viewBox="0 0 177 177"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M116 164V172"
              stroke="#4F4742"
              strokeLinecap="round"
              strokeWidth="10"
            />
            <path
              d="M116 32H121V31.6316L120.946 31.2673L116 32ZM121 167V32H111V167H121ZM120.946 31.2673C120.23 26.4341 118.44 18.7426 113.587 12.1585C108.545 5.31666 100.414 0 88 0V10C97.0862 10 102.289 13.6833 105.538 18.0915C108.977 22.7574 110.437 28.5659 111.054 32.7327L120.946 31.2673ZM88 0C75.0773 0 67.2216 6.82543 62.7924 14.991C58.5379 22.8345 57.3567 31.9633 57.0092 37.6975L66.9908 38.3025C67.31 33.0367 68.3787 25.6655 71.5826 19.759C74.6117 14.1746 79.4227 10 88 10V0Z"
              fill="#A96215"
            />
          </svg>
        </div>
        <div>
          <h2
            className="typo-undertittel"
          >
            <span>
              Innboks
            </span>
          </h2>
          
        </div>
      </div>
      <span
        className="lenkepanel__indikator"
      />
    </a>
  </div>
</div>
`;
